addons/web_editor/static/src/js/wysiwyg_snippets/snippets.editor.js

```diff
@@ -343,6 +345,40 @@ var SnippetEditor = Widget.extend({
 
         return Promise.all(defs);
     },
+    /**
+     * Called when a Mega Menu is removed
+     *
+     * @private
+     */
+    _removeMegaMenu: function () {
```

Comment:
What is all this ? Why are implementing this while it is not asked, probably not wanted and... the feature itself does not even work yet. Please make clean commits to have:
- Ability to enable/disable mega menu on new/old menu items (still not working and it is the very first step to do ...)
- Ability to open the menu in edit mode, drag drop a simple snippet in it and edit it without the menu to close itself on each click

Only when this is perfect and validated we can think about great snippets to design. And only then we can think about if what you implement here is useful or not.

Reply:
This is already what I'm currently working on. I'm not thinking or even building new snippets for now. I've just done multiple commits days ago so re-focus on what you told me in your previous review. Multiple things are working now (or will work in my next push). I assume you did not test it again.

By the way, what is "not asked" here ? Even if I'm not currently working on the expected behavior when a mega menu is removed by the user, obviously something should be done here. I mean, if the user deletes/removes a mega menu, it supposed to remove the mega menu and "reset" the menu item instead of having a broken mega menu (as the user deleted the dropdown).

Reply:
- I obviously tested it again, I would not tell you it is not working if I did not test... :disappointed: The menu closes itself on column resize, on remove, on ... and you still cannot enable a mega menu on a newly created item.
- The user should not be able to remove the mega menu from the edit mode, you should only be able to enable/disable it from the edit menu dialog as this is the simplest thing to do first.

Reply:
Yes, you'll be able to enable/disable a mega menu on a newly created item or "old" item in my next push because I've just done it this morning. Oh, weird. The only thing closes the dropdown is when I delete an element like a column or something, otherwise it seems to work. I'll check this out.

> The user should not be able to remove the mega menu from the edit mode, you should only be able to enable/disable it from the edit menu dialog as this is the simplest thing to do first.

Ok, then. So to make things clear, what is expected when the user is trying to delete the dropdown when in editor mode ? Or what you mean is simply do not be able to delete the dropdown (like no trash icon) ?

As always, thank you for your time ;)